client/grpc-web-core/src/transport.ts

```diff
@@ -0,0 +1,51 @@
+import {Metadata} from "./metadata";
+import {MethodDefinition} from "./service";
+import {ProtobufMessage} from "./message";
+
+export interface TransportOptions {
+    methodDefinition: MethodDefinition<ProtobufMessage, ProtobufMessage>;
+    debug: boolean;
+    url: string;
+    onHeaders: (headers: Metadata, status: number) => void;
+    onChunk: (chunkBytes: Uint8Array, flush?: boolean) => void;
+    onEnd: (err?: Error) => void;
+}
+
+export interface TransportFactory {
+    (options: TransportOptions): Transport;
+}
+
+export interface Transport {
+    sendMessage(msgBytes: Uint8Array): void
+    finishSend(): void
+    cancel(): void
+    start(metadata: Metadata): void
+}
+
+class NullTransport implements Transport {
+    cancel(): void {
+        throw new Error("No transport configured")
+    }
+
+    finishSend(): void {
+        throw new Error("No transport configured")
+    }
+
+    sendMessage(_: Uint8Array): void {
+        throw new Error("No transport configured")
+    }
+
+    start(_: Metadata): void {
+        throw new Error("No transport configured")
+    }
+}
+
+let defaultTransportFactory: TransportFactory = (_: TransportOptions) => new NullTransport();
```

Comment:
as discussed, this should become a "true" global at some point, but let's do that in another PR.

Reply:
Actually, we should make this change in this PR to avoid issues with consuming a webpack'd build; somehting like https://github.com/ungap/global-this seems appropriate.